biz.aQute.bndlib/src/aQute/bnd/cdi/CDIAnnotations.java

```diff
@@ -0,0 +1,219 @@
+			boolean negation = false;
+			if (s.startsWith("!")) {
+				negation = true;
+				s = s.substring(1);
+			}
+			Options option = Options.valueOf(s);
+			if (negation) {
+				options.remove(option);
+			} else {
+				options.add(option);
+			}
+		}
+	}
+
+	@Override
+	public boolean analyzeJar(Analyzer analyzer) throws Exception {
+		Parameters header = OSGiHeader.parseHeader(analyzer.getProperty(Constants.CDIANNOTATIONS, "*"));
+		if (header.size() == 0)
+			return false;
+
+		Parameters optionsHeader = OSGiHeader
+			.parseHeader(analyzer.mergeProperties(Constants.CDIANNOTATIONS_OPTIONS));
+		EnumSet<Options> options = EnumSet.noneOf(Options.class);
+		for (Map.Entry<String, Attrs> entry : optionsHeader.entrySet()) {
+			try {
+				Options.parseOption(entry, options, this);
+			} catch (IllegalArgumentException e) {
+				analyzer.error("Unrecognized %s value %s with attributes %s, expected values are %s",
+					Constants.CDIANNOTATIONS_OPTIONS, entry.getKey(), entry.getValue(), EnumSet.allOf(Options.class));
+			}
+		}
+
+		if (options.isEmpty()) {
+			// set the default mode, if org.osgi.service.cdi.annotations.Bean is
+			// on the classpath set to `annotated_by_bean`, otherwise set to
+			// `none`
+
+			TypeRef typeRef = analyzer.getTypeRefFromFQN("org.osgi.service.cdi.annotations.Bean");
+			if (typeRef != null)
+				options.add(Options.annotated_by_bean);
+			else
+				options.add(Options.none);
+		}
+
+		if (options.contains(Options.none)) {
+			return false;
+		}
+
+		Instructions instructions = new Instructions(header);
+		Collection<Clazz> list = analyzer.getClassspace()
+			.values();
+		List<String> names = new ArrayList<>();
+
+		TreeSet<String> provides = new TreeSet<>();
+		TreeSet<String> requires = new TreeSet<>();
+
+		for (Clazz c : list) {
+			for (Instruction instruction : instructions.keySet()) {
+				if (instruction.matches(c.getFQN())) {
+					if (instruction.isNegated()) {
+						break;
+					}
+					List<BeanDef> definitions = CDIAnnotationReader.getDefinition(c, analyzer, options);
+					if (definitions == null) {
+						break;
+					}
+
+					names.add(definitions.get(0).implementation.getFQN());
+
+					for (BeanDef beanDef : definitions) {
+						if (!beanDef.service.isEmpty()) {
+							int length = beanDef.service.size();
+							String[] objectClass = new String[length];
+
+							for (int i = 0; i < length; i++) {
+								Descriptors.TypeRef tr = beanDef.service.get(i);
+								objectClass[i] = tr.getFQN();
+							}
+							Arrays.sort(objectClass);
+							addServiceCapability(objectClass, provides);
+						}
+					}
+
+					MergedRequirement serviceReqMerge = new MergedRequirement("osgi.service");
+					for (ReferenceDef ref : definitions.get(0).references) {
+						addServiceRequirement(ref, serviceReqMerge);
+					}
+					requires.addAll(serviceReqMerge.toStringList());
+
+					break;
+				}
+			}
+		}
+
+		if (!names.isEmpty()) {
+			addExtenderRequirement(requires, names, CDIAnnotationReader.V1_0);
```

Comment:
Shouldn't this be already handled by the `@RequireCDIExtender` meta-annotation? For DS, I want to leave this out of DS annotation processor and leave it to the requirement annotation processor.

Reply:
I need to attributes to the requirement...

Reply:
Hmmm. We may to to make sure we don't avoid duplicate requirements like we can currently get on DS in #2545.